biz.aQute.bndlib/src/aQute/bnd/build/Workspace.java

```diff
@@ -1185,7 +1188,9 @@ public Project createProject(String name) throws Exception {
 
 		IO.store("#\n#   " + name.toUpperCase()
 			.replace('.', ' ') + "\n#\n", getFile(pdir, Project.BNDFILE));
-		Project p = new Project(this, pdir);
+
+		projects.refresh();
```

Comment:
I did not do this because they you cannot update many projects. Also, getProject() can return null which is not handled. Last, if you want to do this right we should add add/remove project methods.